packages/ui/src/components/editor/panels/Hierarchy/node/index.tsx

```diff
@@ -121,37 +118,23 @@ export const HierarchyTreeNode = (props: HierarchyTreeNodeProps) => {
     setVisibleComponent(entity, !hasComponent(entity, VisibleComponent))
   }
 
-  const onClickToggle = useCallback(
-    (e: MouseEvent) => {
-      e.stopPropagation()
-      if (data.onToggle) data.onToggle(e, entity)
-    },
-    [data.onToggle, node]
-  )
-
-  const onNodeKeyDown = useCallback(
-    (e: KeyboardEvent) => {
-      if (data.onKeyDown) data.onKeyDown(e as any, entity)
-    },
-    [data.onKeyDown, node]
-  )
-
-  const onKeyDownNameInput = useCallback(
-    (e: KeyboardEvent) => {
-      if (e.key === 'Escape') data.onRenameSubmit(entity, null!)
-      else if (e.key === 'Enter') data.onRenameSubmit(entity, (e.target as any).value)
-    },
-    [data.onRenameSubmit, node]
-  )
-
-  const onClickNode = useCallback((e) => data.onClick(e, entity), [node, data.onClick])
-  const onMouseDownNode = useCallback((e) => data.onMouseDown && data.onMouseDown(e, entity), [node, data.onMouseDown])
-
-  const onChangeNodeName = useCallback((e) => data.onChangeName(entity, e.target.value), [node, data.onChangeName])
+  const onClickToggle = (e: React.MouseEvent) => {
```

Comment:
use callback here for optimization

Reply:
Here we are passing these functions directly to the event handlers. Had these functions been dependencies of some `useMemo` or `useEffect`, we could have wrapped them in `useCallback`s. In this case, it would rather been an overhead (due to redundant cache being stored) instead of optimization.

Reply:
Gotcha